app/src/main/java/by/paranoidandroid/serviceexample/services/HardJobIntentService.java

```diff
@@ -0,0 +1,53 @@
+package by.paranoidandroid.serviceexample.services;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.Looper;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+public class HardJobIntentService extends IntentService {
```

Comment:
IntentService is subject to all the background execution limits imposed with Android 8.0 (API level 26). In most cases, you are better off using JobIntentService, which uses jobs instead of services when running on Android 8.0 or higher.

Reply:
But we can't stop JobIntentService immediately, can we? And to meet requirements of the task, we should be able to stop the service by button click...